manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -331,9 +332,8 @@ public void deleteProcessor(String bridgeId, String processorId, String customer
                 processor.getBridge().getId());
     }
 
-    private boolean isProcessorActionable(Processor processor) {
-        // bridge could only be deleted if its in READY or FAILED state
-        return processor.getStatus() == ManagedResourceStatus.READY || processor.getStatus() == ManagedResourceStatus.FAILED;
+    private boolean isManagedResourceActionable(ManagedResource managedResource) {
```

Comment:
very minor question, just curious - do we need to generalise this private method? is not enough by just getting processor param as it is already?

Reply:
It's not a minor question. It's a good spot :smile: 

When making the change I had originally also checked the status of the `Bridge` (`READY` or `FAILED`). That is why I changed the method signature to `isManagedResoureceActionable(..)`. Then I decided to not bother checking the status as it made little sense - there can be no `Processor`'s on a `Bridge` that is not `READY` and there can be no `Processor`'s on a `Bridge` that is `DEPROVISION` or `DELETING`.

However your question has made me re-think. The _old_ implementation threw a `BridgeLifecycleException` to Users if the `Bridge` was not `READY` whereas following my change the method would have returned an empty collection. I have now modified this PR to throw the `BridgeLifecycleException` if the `Bridge` is neither `READY` or `FAILED` that is consistent with how the method used to work.

Reply:
Thanks for the explanation. It makes sense indeed. Good work!